doc/design/diagrams/request_response/client_and_server_options.puml

```diff
@@ -0,0 +1,29 @@
+@startuml
+skinparam backgroundColor #EEEBDC
+
+class ClientOptions {
+  queueCapacity: uint64_t
+  nodeName: NodeName_t
+  connectOnCreate: bool
+  queueFullPolicy: QueueFullPolicy
+  serverTooSlowPolicy: ConsumerTooSlowPolicy
+}
+
+class ServerOptions {
+  queueCapacity: uint64_t
+  nodeName: NodeName_t
+  offerOnCreate: bool
+  clientTooSlowPolicy: ConsumerTooSlowPolicy
+}
+
+enum QueueFullPolicy {
+  BLOCK_PRODUCER,
+  DISCARD_OLDEST_DATA
+}
+
+enum ConsumerTooSlowPolicy {
+  WAIT_FOR_CONSUMER,
+  DISCARD_OLDEST_DATA
+}
```

Comment:
btw, this would replace the current `QueueFullPolicy` and `SubscriberTooSlowPolicy`

Reply:
@elfenpiff @MatthiasKillat @mossmaurice @dkroenke @FerdinandSpitzschnueffler to continue with the implementation I would refactor the stuff above first. This changes the public API, though. Are you okay with this part of the design?

Reply:
@elfenpiff @MatthiasKillat @mossmaurice @dkroenke @FerdinandSpitzschnueffler to continue with the implementation I would refactor the stuff above first. This changes the public API, though. Are you okay with this part of the design?